doc/source/whatsnew/v0.25.0.rst

```diff
@@ -356,6 +356,7 @@ I/O
 - Bug in :func:`read_hdf` not properly closing store after a ``KeyError`` is raised (:issue:`25766`)
 - Bug in ``read_csv`` which would not raise ``ValueError`` if a column index in ``usecols`` was out of bounds (:issue:`25623`)
 - Improved :meth:`pandas.read_stata` and :class:`pandas.io.stata.StataReader` to read incorrectly formatted 118 format files saved by Stata (:issue:`25960`)
+- Fixed bug in loading objects from S3 that contain # characters in the URL (:issue:`25945`)
```

Comment:
use double-back ticks on the ``#``. you can say this is in ``read_*`` routines such as ``:func:read_csv``

Reply:
Fixed.